x-pack/plugins/task_manager/server/polling_lifecycle.ts

```diff
@@ -299,15 +299,16 @@ export function claimAvailableTasks(
           // we can identify the reason
           // if we can - we emit an FillPoolResult error rather than erroring out the wrapping Observable
           // returned by `claimAvailableTasks`
-          if (identifyEsError(ex).includes('cannot execute [inline] scripts')) {
+          if (isEsCannotExecuteScriptError(ex)) {
             logger.warn(
               `Task Manager cannot operate when inline scripts are disabled in Elasticsearch`
             );
             observer.next(asErr(FillPoolResult.Failed));
             observer.complete();
           } else {
+            const esError = identifyEsError(ex);
             // as we could't identify the reason - we'll error out the wrapping Observable too
-            observer.error(ex);
+            observer.error(esError.length > 0 ? esError : ex);
```

Comment:
Yes. Instead of very non descriptive message like we get the full
`server log [10:23:53.592] [error][plugins][taskManager] Failed to poll for work: ResponseError: search_phase_execution_exception`

Reply:
Ah, ok, much better!